src/services/resource.js

```diff
@@ -210,6 +358,364 @@ module.exports = class resourceHelper {
+		}
+		const resourcesDetails = await resourceQueries.resourceList(resourceFilter, ['id'])
+		const resoureId = resourcesDetails.result.map((item) => {
+			return item.id
+		})
+		return resoureId
+	}
+
+	static async findParallelResources(organization_id, resourceTypesInParallelReview = []) {
+		const resourceFilter = {
+			type: {
+				[Op.in]: resourceTypesInParallelReview,
+			},
+			organization_id,
+			review_type: common.REVIEW_TYPE_PARALLEL,
+			status: { [Op.in]: [common.RESOURCE_STATUS_SUBMITTED, common.RESOURCE_STATUS_IN_REVIEW] },
+		}
+		const resourcesDetails = await resourceQueries.resourceList(resourceFilter, ['id'])
+		const resoureId = resourcesDetails.result.map((item) => {
+			return item.id
+		})
+		return resoureId
+	}
+	// fetch all the resource ids from reviews table which is assigned to the reviewer
+	//  and already started reviewer
+	static async findResourcesAssignedToReviewer(organization_ids, reviewer_id) {
+		const reviewsFilter = {
+			reviewer_id,
+			organization_id: { [Op.in]: organization_ids },
+			status: { [Op.in]: common.PAGE_STATUS_VALUES.up_for_review },
+		}
+
+		const reviewsResponse = await reviewsQueries.findAll(reviewsFilter, ['resource_id'])
+		const res = reviewsResponse.map((item) => {
+			return item.resource_id
+		})
+		return res
+	}
+
+	static async findResourcesPickedUpByAnotherReviewer(reviewerId, openToAllResourcesMatchingMyLevel) {
+		// remove all the resouces in sequential review picked up by another reviewer
+		const reviewsFilter = {
+			reviewer_id: { [Op.notIn]: [reviewerId] },
+			status: {
+				[Op.in]: [
+					common.REVIEW_STATUS_INPROGRESS,
+					common.REVIEW_STATUS_CHANGES_UPDATED,
+					common.REVIEW_STATUS_REQUESTED_FOR_CHANGES,
+				],
+			},
+			resource_id: { [Op.in]: openToAllResourcesMatchingMyLevel },
+		}
+		const reviewsResponse = await reviewsQueries.findAll(reviewsFilter, ['resource_id'])
+
+		// push resource ids to resourceIdsToBeRemoved array
+		const resourceIdsToBeRemoved = reviewsResponse.map((item) => item.resource_id)
+		return resourceIdsToBeRemoved
+	}
+
+	// this function fetches review levels from the reviews table for
+	// an organization, roles and resouce type and returns this as an object
+	// Ex : this function returns
+	// {
+	// 	project : 1,
+	// 	observation : 4
+	// }
+	static async fetchResourceReviewLevel(organization_id, userRoleTitles, resourceTypeList) {
+		// fetch review levels according to roles in the organization
+		const reviewLevelDetails = await reviewStagesQueries.findAll(
+			{
+				organization_id,
+				role: {
+					[Op.in]: userRoleTitles,
+				},
+				resource_type: {
+					[Op.in]: resourceTypeList,
+				},
+			},
+			{ attributes: ['resource_type', 'level'], order: [['level', 'ASC']] }
+		)
+		// arrange it as a key-value pair for ease of use
+		const resourceWiseLevels = reviewLevelDetails.reduce((acc, item) => {
+			acc[item.resource_type] = item.level
+
+			return acc
+		}, {})
+
+		return resourceWiseLevels || {}
```

Comment:
@adithyadinesh0412 already defining the {} in the const resourceWiseLevels = reviewLevelDetails.reduce((acc, item) => {
			acc[item.resource_type] = item.level

			return acc
		}, {}) itself so no need of || {} you can just return